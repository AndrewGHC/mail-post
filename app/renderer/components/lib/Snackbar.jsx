// @flow
import React, { Component } from 'react';
import MaterialSnackbar from 'material-ui/Snackbar';
import IconButton from 'material-ui/IconButton';
import CloseIcon from 'material-ui-icons/Close';

type Props = {
  id: string,
  message: string,
};

type State = {
  open: boolean,
};

export default class Snackbar extends Component<Props, State> {
  state = {
    open: false,
  };

  componentWillReceiveProps(nextProps: Props) {
    const { id } = this.props;
    const nextId = nextProps.id;
    if (id !== nextId) this.setState({ open: true });
  }

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { message } = this.props;
    const { open } = this.state;
    return (
      <MaterialSnackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        open={open}
        autoHideDuration={6000}
        onClose={this.handleClose}
        SnackbarContentProps={{
          'aria-describedby': 'message-id',
        }}
        message={<span id="message-id">{message}</span>}
        action={[
          <IconButton
            key="close"
            aria-label="Close"
            color="inherit"
            onClick={this.handleClose}
          >
            <CloseIcon />
          </IconButton>,
        ]}
      />
    );
  }
}
