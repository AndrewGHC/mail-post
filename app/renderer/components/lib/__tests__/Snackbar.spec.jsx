import React from 'react';
import { shallow } from 'enzyme';
import MaterialSnackbar from 'material-ui/Snackbar';

import Snackbar from '../Snackbar';

describe('Snackbar', () => {
  it('renders a material ui snackbar', async () => {
    const wrapper = shallow(<Snackbar />);
    expect(wrapper.find(MaterialSnackbar).exists()).toBe(true);
  });

  it('matches snapshot', async () => {
    const wrapper = shallow(<Snackbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
