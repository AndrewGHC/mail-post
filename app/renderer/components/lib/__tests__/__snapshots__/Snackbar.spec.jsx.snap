// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar matches snapshot 1`] = `
<WithStyles(Snackbar)
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <WithStyles(IconButton)
        aria-label="Close"
        color="inherit"
        onClick={[Function]}
      >
        <pure(Close) />
      </WithStyles(IconButton)>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="message-id"
    />
  }
  onClose={[Function]}
  open={false}
/>
`;
